Adodbapi missing files

vernondcole opened the pull request:
This PR does much more than the title suggests. 

With this PR, the pywin32/adodbapi subfolder becomes the definitive source for adodbapi. Therefore, the source code can never get out of sync again.

The changes here include:
- two bug fixes
- removal of support for Python 2.6 (with simplifications for testing)
- (unsupported) remote access database connection feature
- test script set to use Internet available database servers for MS-sql, postgreSQL, and MariaDB.
- documentation converted from .odt to .md format.

Query: with the addition of my three online database servers, it is possible to run a full "--all" test from anywhere, but that adds about five minutes, per pass, to the unit testing for USA based users. I have, therefore set the default test to run on only a local Jet database. It is not a complete test, but takes only around three seconds. I am working on a subsystem which will spin up Vagrant virtual machines for building and testing pywin32. If that were in place, the long Internet delay in testing would go away, and --all tests would be viable.  Did I pick the correct optimization for the present moment?

Explanation: Testing of adodbapi has always been problematical because of the need for test databases. Microsoft has been kind enough to grant me an open source developer's free account which includes the ability to run a small VM on the azure cloud. Using that, I have spun up an Ubuntu-based SQL Server instance that we can use for testing. I have also opened up access on non-standard ports for MySQL and PostgreSQL servers on small VMs that I fund myself in France and California. None of these machines have enough capacity to make it worthwhile for anyone to use them for anything useful, nor are they mission critical for other uses.

mhammond commented:
I think you did pick the sweet-spot for testing, particularly if the servers being hit are ones not really expecting more traffic. That said though, the reality is that I'm probably not going to work out how to run the more exhaustive tests regularly, so there's a bit of a risk they will not get run. However, we can iterate on that over time.

The changes here look fine. We should get something in CHANGES.txt too, explaining a bit of the motivation etc. If you could add that and then let me know when you think it's ready to merge, I'll pull the pin!

vernondcole commented:
Okay. I added a short paragraph in CHANGES.txt and a bit to adodbapi/readme.txt.
It looks like I have fixed enough things that the failures in the CI tests are no longer caused by me.
I don't have a build environment so that I can actually create a test binary, which makes me nervous,
now that I am officially a Q.A. engineer. 
I have created a [Salt](https://docs.saltstack.com/en/latest/)  system for provisioning [build machines](https://github.com/salt-bevy/), and I intend to use the pywin32 build environment as an example script there. Then we could add a link to it, and some simple instructions, and anybody could spin up a build machine quite easily, so they can test or customize on their own environment. 
I don't have that yet, so for now -- everything I *can* test works, so, yes, pull the pin.

mhammond commented:
Awesome, thanks!
